### DISCUSSION TITLE
What is the difference between clrDgItemsOf and clrDgItems?

### DISCUSSION THREAD

**Discussion by kumar-tadepalli** [vmware-clarity/ng-clarity#797] [Q&A] [answered]
I'm going through the datagrid documentation there are two directives clrDgItemsOf clrDgItems
![image](https://github.com/vmware-clarity/ng-clarity/assets/12848838/a2adb47b-16d6-426c-8b44-8fdd889428bd)

I didn't understand when/At what scenarios we need clrDgItemsOf 

Some one provide the with example code?

**Comment by kevinbuhmann** [comment, marked as answer]
No difference. It's the same directive. The selector is just Angular syntax for an iterator structural directive.

This code:

```html
<clr-dg-row *clrDgItems="let item of items" [clrDgItem]="item">
  ...
</clr-dg-row>
```

is syntactic sugar (or shorthand) for:

```html
<ng-template clrDgItems [clrDgItemsOf]="items" let-item>
  <clr-dg-row [clrDgItem]="item">
    ...
  </clr-dg-row>
</ng-template>
```

https://angular.io/guide/structural-directives
